internal/provider/kubernetes/gateway.go

```diff
@@ -67,12 +67,26 @@ func newGatewayController(mgr manager.Manager, cfg *config.Server, su status.Upd
 	if err := c.Watch(
 		&source.Kind{Type: &gwapiv1b1.Gateway{}},
 		&handler.EnqueueRequestForObject{},
-		predicate.NewPredicateFuncs(r.hasMatchingController),
+		predicate.NewPredicateFuncs(r.gatewayHasMatchingController),
 	); err != nil {
 		return err
 	}
 	r.log.Info("watching gateway objects")
 
+	// Only enqueue GatewayClass objects that match this Envoy Gateway's controller name.
+	if err := c.Watch(
+		&source.Kind{Type: &gwapiv1b1.GatewayClass{}},
+		&handler.EnqueueRequestForObject{},
```

Comment:
I *think* you want to use r.getGatewaysForClass here as a map func -- so watch GatewayClasses with a matching controller string, then trigger reconcile requests for the Gateways of that class.

Otherwise, getGatewaysForClass isn't being used anywhere